### ISSUE TITLE
Add tests on FNO + DeepONet forward pass, gradients, and training

### ISSUE THREAD

**Issue by ChrisRackauckas** [SciML/OperatorLearning.jl#26]
The tests don't seem to cover the use and training of the operators, just a few properties. It would be good to get a few integration tests.

**Comment by ba2tripleO** [comment]
Do these tests need to run on gpu?

**Comment by pzimbrod** [comment]
Yeah I think that makes sense at least for a subset of those, we also track that in a separate issue (#27).

Flux.jl also contains CUDA in the [tests](https://github.com/FluxML/Flux.jl/blob/master/test/runtests.jl#L52-L58) and since this package also exports layers and is based on Flux quite a bit, I would more or less stick to this.

**Comment by ChrisRackauckas** [comment]
I would start by making the tests on CPU though, then create equivalent tests on GPU. GPU will have a separate set of issues and it would be easier to catch/debug the CPU issues (usually correctness) first.
